llvm/lib/CodeGen/WindowScheduler.cpp

```diff
@@ -192,22 +192,37 @@ bool WindowScheduler::initialize() {
     return false;
   }
   // Check each MI in MBB.
-  SmallVector<Register, 8> PhiDefs;
+  SmallSet<Register, 8> PrevDefs;
+  SmallSet<Register, 8> PrevUses;
+  auto IsLoopCarried = [&](MachineInstr &Phi) {
+    // Two cases are checked here: (1)The virtual register defined by the
+    // preceding phi is used by the succeeding phi;(2)The preceding phi uses the
+    // virtual register defined by the succeeding phi.
+    auto Def = Phi.getOperand(0);
+    if (Def.isReg()) {
```

Comment:
Def must be a register. Don't try to handle invalid MIR

Reply:
Updated